crates/sui-verifier/src/entry_points_verifier.rs

```diff
@@ -199,6 +200,14 @@ fn verify_param_type(
     function_type_args: &[AbilitySet],
     param: &SignatureToken,
 ) -> Result<(), String> {
+    if is_mutable_clock(view, param) {
```

Comment:
Should we maybe check this in the transaction input checker? I worry that this feels a bit strange, but maybe I'm just annoyed by the fact that I would have to replicate this for public functions in programmable transactions :P

Reply:
Recording our offline chat: This will also be checked in `transaction_input_checker`, this is just an additional safety net.

Reply:
>this is just an additional safety net.

All of the safety nets!!!

Reply:
Speaking of all of the safety nets, we're kind of lucky that we can do this for `Clock`, but if throughput grows to the point where RWLocks are necessary for the performance of transactions on arbitrary shared objects, we may end up in trouble if people are allowed to arbitrarily take the write lock on a shared object when they only need to read it (it's a denial of service vector), so we should consider adding yet another safety net that you can only take the writer lock for a `&mut` parameter.

Reply:
how easy/difficult would that be to do?

Also, how are we approaching limiting who can do things like increment the clock? Are we using a non-public entry function to do this so that only the validators can increment it?

Reply:
> how easy/difficult would that be to do?

I think it's doable -- it would need support during validator signing to load the method signature and check that mutable shared object parameters only flow into mutable reference parameters, and then support in the SDK to load the signature and set the `mutable` flags accordingly.

There was initially some scepticism around incurring the cost of loading the package during signing, especially in the case of single owner transactions, but we can just skip this step entirely if there are no shared objects, so I'm not too worried there.

> Also, how are we approaching limiting who can do things like increment the clock? Are we using a non-public entry function to do this so that only the validators can increment it?

Yep, this is implemented in #8055 which introduces a new system transaction for setting up things in a checkpoint, including setting the timestamp for the clock.